qa/restart-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/SemanticSearch.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.bwc;
+
+import com.carrotsearch.randomizedtesting.RandomizedTest;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.ArrayList;
+
+
+import static org.opensearch.neuralsearch.TestUtils.NODES_BWC_CLUSTER;
+
+import static org.opensearch.neuralsearch.TestUtils.getModelId;
+import static org.opensearch.neuralsearch.TestUtils.TEXT_EMBEDDING_PROCESSOR;
+import org.opensearch.neuralsearch.query.NeuralQueryBuilder;
+
+
+public class SemanticSearch extends AbstractRestartUpgradeRestTestCase{
+
+    private static final String PIPELINE_NAME = "nlp-pipeline";
+    private static String DOC_ID = "0";
+    private static final String TEST_FIELD = "passage_text";
+    private static final String TEXT= "Hello world";
+
+    //Test restart-upgrade Semantic Search
+    //Create Text Embedding Processor, Ingestion Pipeline and add document
+    //Validate process , pipeline and document count in restart-upgrade scenario
+    public void testSemanticSearch() throws Exception{
+        waitForClusterHealthGreen(NODES_BWC_CLUSTER);
+
+        if (isRunningAgainstOldCluster()){
+            String modelId= uploadTextEmbeddingModel();
+            loadModel(modelId);
+            createPipelineProcessor(modelId,PIPELINE_NAME);
+            createIndexWithConfiguration(
+                    testIndex,
+                    Files.readString(Path.of(classLoader.getResource("processor/IndexMappings.json").toURI())),
+                    PIPELINE_NAME
+            );
+            addDocument(testIndex, DOC_ID,TEST_FIELD,TEXT);
+        }else {
+            Map<String,Object> pipeline= getIngestionPipeline(PIPELINE_NAME);
+            assertNotNull(pipeline);
+            String modelId=getModelId(pipeline, TEXT_EMBEDDING_PROCESSOR);
+            validateTestIndex(modelId);
+            deletePipeline(PIPELINE_NAME);
+            deleteModel(modelId);
+            deleteIndex(testIndex);
+        }
+    }
+
+
+    private void validateTestIndex(String modelId) throws Exception {
```

Comment:
nit: for tests you can use SneakyThrows annotation instead of putting exact Exception to the method signature

Reply:
For all tests it is not applicable as there is an issue with lombok. It cannot rectify checked exception. There is checked exception (URI syntax exception) coming from [here](https://github.com/opensearch-project/neural-search/pull/515/files#diff-e4fb9812973229e1ebaad95042281ab554d59b3c3d16155af075823fe29b8caaR41). And If If add @Sneakythrows then I get below error 
```
                                                                                                                           ^
/Users/varunudr/neuralsearch/neural-search/qa/restart-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/SemanticSearch.java:93: error: unreported exception URISyntaxException; must be caught or declared to be thrown
        String requestBody = Files.readString(Path.of(classLoader.getResource("processor/PipelineConfiguration.json").toURI()));

```

Reply:
got it, no issues